packages/convex-helpers/react/sessions.ts

```diff
@@ -66,25 +74,46 @@ type SessionArgsArray<Fn extends SessionFunction<"mutation" | "action">> =
  *  - localStorage is shared between tabs, but not browser profiles.
  * @param storageKey - Key under which to store the session ID in the store
  * @param idGenerator - Function to return a new, unique session ID string. Defaults to crypto.randomUUID
+ * @param ssrFriendly - Returns SSR_DEFAULT on the first pass, so hydration
+ *   doesn't mismatch and an ID isn't generated server-side.
+ *   useSessionQuery will skip queries until there is a real value.
+ *   However, if you're using useSessionId, consider checking for SSR_DEFAULT.
+ *   Defaults to false, where it will always return a valid id.
  * @returns A provider to wrap your React nodes which provides the session ID.
  * To be used with useSessionQuery and useSessionMutation.
  */
 export const SessionProvider: React.FC<{
   useStorage?: UseStorage<SessionId>;
   storageKey?: string;
   idGenerator?: () => string;
+  ssrFriendly?: boolean;
   children?: React.ReactNode;
-}> = ({ useStorage, storageKey, idGenerator, children }) => {
+}> = ({ useStorage, storageKey, idGenerator, ssrFriendly, children }) => {
   const storeKey = storageKey ?? "convex-session-id";
-  const idGen = idGenerator ?? crypto.randomUUID.bind(crypto);
-  const initialId = useMemo(() => idGen() as SessionId, []);
+  function idGen() {
+    // On the server, crypto may not be defined.
+    return (idGenerator ?? crypto.randomUUID.bind(crypto))() as SessionId;
+  }
   // Get or set the ID from our desired storage location.
   const useStorageOrDefault = useStorage ?? useSessionStorage;
-  const [sessionId, setSessionId] = useStorageOrDefault(storeKey, initialId);
+  const [sessionId, setSessionId] = useStorageOrDefault(
+    storeKey,
+    ssrFriendly ? SSR_DEFAULT : idGen()
+  );
+
+  const [initial, setInitial] = useState(true);
+  if (ssrFriendly) {
```

Comment:
against the rules but should be constant

Reply:
You can also put the if condition inside the useEffect